File: 34. Find First and Last Position of Element in Sorted Array/Solution.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        valid,tar_cnt = False, 0
        for i in nums:
            if i==target:
                valid = True
                tar_cnt += 1
        if not valid:
            return [-1,-1]
        else:
            if tar_cnt==1:
                for i,n in enumerate(nums):
                    if n==target:
                        return [i,i]
            start,end = -1, -1
            for i,n in enumerate(nums):
                if n==target:
                    if start==-1:
                        start=i
                    else:
                        end=i
            return [start,end]

File: 34. Find First and Last Position of Element in Sorted Array/Solution.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        bool valid = false;
        int tar_cnt = 0;
        for(int i=0; i<nums.size(); i++){
            if(nums[i]==target){
                valid = true;
                tar_cnt++;
            }
        }
        if(!valid){
            return {-1,-1};
        }else{
            if(tar_cnt==1){
                for(int i=0; i<nums.size(); i++){
                    if(nums[i]==target){
                        return {i,i};
                    }
                }
            }
            int start=-1, end=-1;
            for(int i=0; i<nums.size(); i++){
                if(nums[i]==target){
                    if(start==-1){
                        start = i;
                    }else{
                        end = i;
                    }
                }
            }
            return {start,end};
        }
    }
};
